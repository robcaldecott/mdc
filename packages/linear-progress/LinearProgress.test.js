import React from "react";
import { LinearProgress } from ".";
import renderer from "react-test-renderer";

it("renders indeterminate", () => {
  expect(renderer.create(<LinearProgress indeterminate />)).toMatchSnapshot();
});

it("renders indeterminate reversed", () => {
  expect(
    renderer.create(<LinearProgress indeterminate reversed />)
  ).toMatchSnapshot();
});

it("renders determinate", () => {
  expect(
    renderer.create(<LinearProgress progress={0.25} buffer={0.25} />)
  ).toMatchSnapshot();
});

it("renders determinate reversed", () => {
  expect(
    renderer.create(<LinearProgress progress={0.25} buffer={0.25} reverseed />)
  ).toMatchSnapshot();
});

it("renders closed", () => {
  expect(renderer.create(<LinearProgress closed />)).toMatchSnapshot();
});

it("renders with a custom class", () => {
  expect(
    renderer.create(<LinearProgress className="custom" />)
  ).toMatchSnapshot();
});
