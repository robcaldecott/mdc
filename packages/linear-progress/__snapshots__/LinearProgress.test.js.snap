// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders closed 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--closed"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "MSTransform": "scaleX(1)",
        "MozTransform": "scaleX(1)",
        "OTransform": "scaleX(1)",
        "WebkitTransform": "scaleX(1)",
        "transform": "scaleX(1)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "MSTransform": "scaleX(undefined)",
        "MozTransform": "scaleX(undefined)",
        "OTransform": "scaleX(undefined)",
        "WebkitTransform": "scaleX(undefined)",
        "transform": "scaleX(undefined)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`renders determinate 1`] = `
<div
  className="mdc-linear-progress"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "MSTransform": "scaleX(0.25)",
        "MozTransform": "scaleX(0.25)",
        "OTransform": "scaleX(0.25)",
        "WebkitTransform": "scaleX(0.25)",
        "transform": "scaleX(0.25)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "MSTransform": "scaleX(0.25)",
        "MozTransform": "scaleX(0.25)",
        "OTransform": "scaleX(0.25)",
        "WebkitTransform": "scaleX(0.25)",
        "transform": "scaleX(0.25)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`renders determinate reversed 1`] = `
<div
  className="mdc-linear-progress"
  reverseed={true}
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "MSTransform": "scaleX(0.25)",
        "MozTransform": "scaleX(0.25)",
        "OTransform": "scaleX(0.25)",
        "WebkitTransform": "scaleX(0.25)",
        "transform": "scaleX(0.25)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "MSTransform": "scaleX(0.25)",
        "MozTransform": "scaleX(0.25)",
        "OTransform": "scaleX(0.25)",
        "WebkitTransform": "scaleX(0.25)",
        "transform": "scaleX(0.25)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`renders indeterminate 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--indeterminate"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "MSTransform": "scaleX(1)",
        "MozTransform": "scaleX(1)",
        "OTransform": "scaleX(1)",
        "WebkitTransform": "scaleX(1)",
        "transform": "scaleX(1)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "MSTransform": "scaleX(1)",
        "MozTransform": "scaleX(1)",
        "OTransform": "scaleX(1)",
        "WebkitTransform": "scaleX(1)",
        "transform": "scaleX(1)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`renders indeterminate reversed 1`] = `
<div
  className="mdc-linear-progress mdc-linear-progress--indeterminate mdc-linear-progress--reversed"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "MSTransform": "scaleX(1)",
        "MozTransform": "scaleX(1)",
        "OTransform": "scaleX(1)",
        "WebkitTransform": "scaleX(1)",
        "transform": "scaleX(1)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "MSTransform": "scaleX(1)",
        "MozTransform": "scaleX(1)",
        "OTransform": "scaleX(1)",
        "WebkitTransform": "scaleX(1)",
        "transform": "scaleX(1)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;

exports[`renders with a custom class 1`] = `
<div
  className="mdc-linear-progress custom"
  role="progressbar"
>
  <div
    className="mdc-linear-progress__buffering-dots"
  />
  <div
    className="mdc-linear-progress__buffer"
    style={
      Object {
        "MSTransform": "scaleX(1)",
        "MozTransform": "scaleX(1)",
        "OTransform": "scaleX(1)",
        "WebkitTransform": "scaleX(1)",
        "transform": "scaleX(1)",
      }
    }
  />
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
    style={
      Object {
        "MSTransform": "scaleX(undefined)",
        "MozTransform": "scaleX(undefined)",
        "OTransform": "scaleX(undefined)",
        "WebkitTransform": "scaleX(undefined)",
        "transform": "scaleX(undefined)",
      }
    }
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
  <div
    className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
  >
    <span
      className="mdc-linear-progress__bar-inner"
    />
  </div>
</div>
`;
